backend/services/implementations/testSessionService.ts

```diff
@@ -187,6 +197,69 @@ class TestSessionService implements ITestSessionService {
 
     return testSessionDtos;
   }
+
+  /*
+  * computeTestGrades computes the breakdown and score of a given result and returns the ResultDTO
+  */
+  async computeTestGrades(result: NewResultDTO, testId: string): Promise<ResultDTO> {
+    let resultDto: ResultDTO;
+    let questionsCorrect: number = 0; // the number of questions the student gets correct
+    let studentAnswers: number[] = result.answers; // the array of the student's raw answers
+    let updatedScore: number = 0.00; // the final score that will be returned
+    let updatedBreakdown: boolean[] = []; // the final breakdown that will be returned
+
+    try {
+      const test: TestResponseDTO = await this.testService.getTestById(testId);
+
+      // check if the number of test questions matches the number of student answers
+      if (test.questions.length != studentAnswers.length) {
```

Comment:
I don't know if this will even be a possible case, but I have it here anyways

Side note: Is it possible for a student to leave a question blank? If so, should we check if the answer=null or smth

Reply:
Hmm,  I don't think so. If a student left the answer blank, I think best practice would be to populate from the frontend with a `null` value. So, given this, I'm not sure if `test.questions.length` would ever vary from the `studentAnswers.length`? Let me know if I'm missing a case here!

Reply:
Wait so, on backend would we store smth like `studentAnswers=[10, null, 20]` if second question is blank

Reply:
I think so. Let me know if you think there is a better alternative! And if we do this, we should have `null` checks in the `computeGrades` function then as well, I think.

Reply:
i've updated the interfaces! should work with the current logic since we directly check for strict equality `studentAnswers[i] === actualAnswer`